### ISSUE TITLE
Read some documents metadata from Git repo

### ISSUE THREAD

**Issue by OndraZizka** [jbake-org/jbake#496]
JBake is probably often used on top of a Git (or other SCM) repo.

This can be leveraged to acquire these information automatically:

- document creation date
- document last update date
- document author(s)
- related documents (those changed in the same commit)
- perhaps commit comments could be used for something too 

This would be quite useful. I dare to say, better than getting the creation date from the filesystem, where it's almost impossible to keep unchanged in the long term.

How do you like the idea? It shouldn't be too hard, there are lightweight Java libs for Git metadata reading.

**Comment by ancho** [comment]
I like that idea. It should be relative easy todo. I like the groovy library [grgit](https://github.com/ajoberstar/grgit)
it uses [jgit](https://www.eclipse.org/jgit/) under the hood. Works like a charm. We use it in the gradle build to consume the example projects to bundle them for the distribution.

**Comment by jonbullock** [comment]
Yeah I like that too, could be a perfect candidate for an plugin/extension.
